bears/jinja2/Jinja2Bear.py

```diff
@@ -0,0 +1,335 @@
+            if label is None and line_number != start_in_line:
+                diff = generate_label_diff(
+                    file, filename, line, line_number, m, expected_label)
+                yield Result.from_values(
+                    origin=self,
+                    message="Unlabeled control end tag",
+                    file=filename,
+                    line=line_number,
+                    column=m.start(0) + 1,
+                    end_line=line_number,
+                    end_column=m.end(0) + 1,
+                    diffs=diff)
+            elif label != expected_label and line_number != start_in_line:
+                diff = generate_label_diff(
+                    file, filename, line, line_number, m, expected_label)
+                yield Result.from_values(
+                    origin=self,
+                    message="End tag label does not match expected label",
+                    file=filename,
+                    line=line_number,
+                    column=m.start("label") + 1,
+                    end_line=line_number,
+                    end_column=m.end("label") + 1,
+                    diffs=diff)
+
+    def check_for_empty_control_stack(self, filename):
+        '''
+        Checks if the control stack is empty and yields a Result for each
+        item left on the stack.
+
+        :param filename: The name of the file currently being inspected
+        '''
+        if len(self.control_stack) > 0:
+            for (_, line_number) in self.control_stack:
+                yield Result.from_values(
+                    origin=self,
+                    message="Control start tag has no corresponding end",
+                    file=filename,
+                    line=line_number,
+                    severity=RESULT_SEVERITY.MAJOR)
+
+    def run(self,
+            filename,
+            file,
+            variable_spacing: int=1,
+            control_spacing: int=1):
+        '''
+        Check Jinja2 templates for syntax, formatting and documentation issues.
+        The following aspects are being looked at:
+
+        * Variable spacing:    Variable tags should be padded with one space on
+                               each side, like this: ``{{ var_name }}``. This
+                               can be set to any number of spaces via the
+                               setting variable_spacing.
+                               Malformatted variable tags are detected and
+                               fixes suggested.
+        * Control spacing:     Like variable spacing, but for control blocks,
+                               i.e. if and for constructs. Looks at both start
+                               and end block.
+        * Control labels:      It is good practise to label the end of an if or
+                               for construct with a comment equal to the
+                               content of the start, like so::
```

Comment:
No, it denotes the start of a [literal block](http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html#literal-blocks)